internal/suite/setup.go

```diff
@@ -104,8 +104,8 @@ func Setup(isUpgrade bool, clusterBuilder cb.ClusterBuilder, clusterReadyFns ...
 							cl.Status.Phase,
 							cl.Status.InfrastructureReady,
 							cl.Status.ControlPlaneReady,
-							ptr.Deref(cl.Status.FailureReason, ""),
-							ptr.Deref(cl.Status.FailureMessage, ""),
+							ptr.Deref(cl.Status.FailureReason, ""),  //nolint:staticcheck // Ignore SA1019 this field is marked as deprecated.
```

Comment:
This might be interesting, upstream seem to have dropped some status fields in the future, see https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20240916-improve-status-in-CAPI-resources.m